### DISCUSSION TITLE
Title and subtitle are not showing.

### DISCUSSION THREAD

**Discussion by Premal-Khetani** [LeoNatan/LNPopupController#517] [General] [answered]
I have used the code in the latest swift project. below is the code. 

I am not able to see the bar's title and anything over there. 

> `let appearance = LNPopupBarAppearance.init()`
> `appearance.titleTextAttributes = [.font: UIFont(name: "Avenir-Black", size: 14)!, .foregroundColor: UIColor.yellow]`
> `appearance.subtitleTextAttributes = [.font: UIFont(name: "Avenir-Black", size: 12)!, .foregroundColor: UIColor.green]`
> `appearance.backgroundEffect = UIBlurEffect(style: .systemMaterialDark)`
> `appearance.marqueeScrollEnabled = true`
> `tabbarController?.popupBar.standardAppearance = appearance`
> `tabbarController?.popupItem.standardAppearance = appearance`
> `self.popupBar.popupItem?.title = "test"`
> `self.popupItem.title = "going on."`
> ` self.popupItem.subtitle = "testing..."`

![Screenshot at Sep 27 13-56-14](https://user-images.githubusercontent.com/13691970/192474472-a8921fb3-a143-4400-989b-fc3746cd33c5.png)

**Comment by LeoNatan** [comment, marked as answer]
You are likely setting the title of the incorrect item. See example project.

**Reply by Premal-Khetani** [reply]
I have tried to write it down in viewDidload() and in the method from where the popup is going to open. but it's not working. Can you please help me here?

**Reply by LeoNatan** [reply]
I am not familiar with your app or code. How can I help?

**Reply by Premal-Khetani** [reply]
let me share the code. can you tell me when & where i can set the title of a popup?

**Reply by LeoNatan** [reply]
You need to read the docs. You need to modify the popup item of the controller that **is being presented**, not the one that is the container.

**Reply by Premal-Khetani** [reply]
ok let me prepare a demo and check and share it with you.

**Reply by LeoNatan** [reply]
There is already a demo project in the repo. I don’t need more demos.

**Reply by Premal-Khetani** [reply]
but that is more confusing for me as I cannot set the title by checking and following the same demo. I have read all the articles and https://github.com/LeoNatan/LNPopupController/discussions/516 this link also. I tried with the same thing but cannot see the title bar title and subtitle. I did inspect the elements but was still not able to figure out how could be this solved.

**Reply by LeoNatan** [reply]
See here: https://github.com/LeoNatan/LNPopupController/blob/489e8b0582b3b2421829603a02f89e89e4803306/LNPopupControllerExample/LNPopupControllerExample/DemoViewController.m#L215

**Reply by Premal-Khetani** [reply]
i have used the same way as it's shown in the given link. 

![Screenshot at Sep 27 16-28-37](https://user-images.githubusercontent.com/13691970/192508313-5b7078c3-5eb2-4ce0-9a0b-b5c4009cd089.png)

**Reply by LeoNatan** [reply]
Your screenshot looks good from a quick look

**Reply by Premal-Khetani** [reply]
But I am not getting why it's not working right now. :/

**Reply by Premal-Khetani** [reply]
any help with this? 

![image](https://user-images.githubusercontent.com/13691970/192516977-c2a60a6f-78fb-416b-ab61-5efecc213c16.png)

**Reply by Premal-Khetani** [reply]
@LeoNatan any update mate? any way that it could be resolved?

**Reply by iDevelopper** [reply]
On the view hierarchy the popup bar is a descendant of UINavigationController, not PKTabBarController.

**Reply by Premal-Khetani** [reply]
@iDevelopper  so do I need to set the title of that navigation bar?

**Reply by iDevelopper** [reply]
No, it looks like the popup bar was presented with the navigation controller.

**Reply by Premal-Khetani** [reply]
so I am not able to figure it out. which bar's title and subtitle I should add so that It can be visible. ?

**Reply by iDevelopper** [reply]
Do you present the popup bar elsewhere in your code?

**Reply by Premal-Khetani** [reply]
Nope only here.

**Reply by Premal-Khetani** [reply]
@iDevelopper @LeoNatan hey any update on this? I am not able to figure out which one to set. I am trying all the ways but still have not been able to display the title and subtitle.

**Reply by LeoNatan** [reply]
You keep asking for "updates". What exactly do you want us to update you on? You have a bug somewhere, so debug it and figure it out. We don't have access to your code, nor will be going over your project to figure out your bug. If you are able to reproduce the issue in a demo project, please do so and attach here. If it works in a small demo project, debug your own project and figure out the issue.
